src/services/initializer.ts

```diff
@@ -0,0 +1,117 @@
+import { CLIError } from "@oclif/errors"
+import * as cp from "node:child_process"
+import * as fs from "node:fs"
+import * as path from "node:path"
+import { BundleDescriptor } from "../models/bundle-descriptor"
+
+const ALLOWED_BUNDLE_NAME_REGEXP = /^[\w-]+$/
+
+export interface InitializerOptions {
+  parentDirectory: string
+  name: string
+  version: string
+}
+
+/** Handles the scaffolding of a project bundle */
+export default class BundleProjectInitializer {
+  private readonly options: InitializerOptions
+
+  constructor(options: InitializerOptions) {
+    this.options = options
+  }
+
+  public async performScaffolding(): Promise<void> {
+    if (!ALLOWED_BUNDLE_NAME_REGEXP.test(this.options.name)) {
+      throw new CLIError(
+        `"${this.options.name}" is not a valid bundle name. Only alphanumeric characters, underscore and dash are allowed`
+      )
+    }
+
+    this.createBundleDirectories()
+    this.createBundleDescriptor()
+    this.createDockerfile()
+    this.createGitignore()
+    this.createConfigJson()
+    this.initGitRepo()
+  }
+
+  private createBundleDirectories() {
+    const bundleDir = this.getBundleDirectory()
+
+    try {
+      fs.accessSync(this.options.parentDirectory, fs.constants.W_OK)
+    } catch {
+      throw new CLIError(
+        `Directory ${this.options.parentDirectory} is not writable`
+      )
+    }
+
+    if (fs.existsSync(bundleDir)) {
+      throw new CLIError(`Directory ${bundleDir} already exists`)
+    }
+
+    fs.mkdirSync(bundleDir)
+    fs.mkdirSync(this.getBundleFilePath(".ent"))
+    fs.mkdirSync(this.getBundleFilePath(".ent", "output"))
+    fs.mkdirSync(this.getBundleFilePath("microservices"))
+    fs.mkdirSync(this.getBundleFilePath("microfrontends"))
+    fs.mkdirSync(this.getBundleFilePath("epc"))
+  }
+
+  private createBundleDescriptor() {
```

Comment:
It's worth to move this logic to a dedicated service in order to implement the CRUD there. Wdyt?

Reply:
Yes, it could be useful for update operations done by other classes. I'm adding it